### ISSUE TITLE
Backstage not rendering <iframe> element in Docs

### ISSUE THREAD

**Issue by bshore** [backstage/backstage#7710]
We're using `techdocs-cli` to build our `.md` documentation files and publishing to S3. When viewing the Documentation locally using the `spotify/techdocs` Docker image our `<iframe>` renders. When we view our docs through a deployed Backstage instance though, the `<iframe>` does not render.

I've been looking into this and I believe it may have something to do with Shadow DOM. We have `https://lucid.app` as an allowed `frame-ancestor` in our Backstage `app-config.yaml` Content Security Policy, and I'm not 100% if that is passed to the Shadow DOM of our built documentation, or if it even needs to be passed?

## Expected Behavior

When viewing documentation in Backstage, an embedded Lucid Chart `<iframe>` should render if it is allowed by the CSP.

```html
<div style="width: 640px; height: 480px; margin: 10px; position: relative;">
<iframe allowfullscreen frameborder="0" style="width:640px; height:480px" src="https://lucid.app/documents/embeddedchart/xxxxxxxxxxxxx" id="xxxxxxx"></iframe>
</div>
```

## Current Behavior

When viewing our documentation in Backstage if we inspect element on the Shadow DOM, our `<div>` is present, but the `<iframe>` is missing. We've verified in our S3 and the `<iframe>` is present in the `index.html` of our built documentation, but for some reason is getting stripped or cut out once it is in Backstage.
```html
<div style="width: 640px; height: 480px; margin: 10px; position: relative;">
</div>
```

## Steps to Reproduce

1. Write some `.md` file that has an html block similar to the `<iframe>` example above.
2. View that documentation locally through the `spotify/techdocs` Docker image
3. View the same documentation in Backstage (once docs have been built as `html`, and being pulled in as a Shadow DOM)

## Context

Some of our teams have some complex Lucid Charts explaining their processes, infrastructure, etc. We would like to allow them to use Lucid's embeddable `<iframe>` code for their documentation in Backstage.

## Your Environment

Backend Backstage Dockerfile Base: `node:14-buster-slim`
Edit: We are using [release-2021-10-13](https://github.com/backstage/backstage/releases/tag/release-2021-10-13)

OS:   Darwin 20.6.0 - darwin/x64
node: v14.8.0
yarn: 1.22.4
cli:  0.7.15 (installed)

Browser: Latest Chrome

**Comment by emmaindal** [comment]
Building on ^

Yes I feel like having a way to configure what tags to allow, with the default to not support iframes but possibility to change that could be a path moving forward. And with recommendations why not to support certain tags maybe?

**Comment by dnsaoki2** [comment]
hi @iamEAP , how are you? i saw the issue was put in TODO does it mean you guys will be working on it soon or is it something we can suddenly collaborate on?

**Comment by iamEAP** [comment]
We could be a lot better about our project board 😅

In this case, just to note that it's something we are actively thinking about on the team and will likely address soon (or, if you're interested in collaborating/contributing--we have time allotted to discuss/etc)

**Comment by dnsaoki2** [comment]
@iamEAP we are interested in contributing. Do you already have any suggestions?

**Comment by iamEAP** [comment]
@dnsaoki2 I'd say @emmaindal's recommendation above would be a great place to start.

Likely introducing a new config key in techdocs whose value is an array of strings representing extra tags to allow? And then modifying the sanitizer Emma linked to use that config.

**Comment by emmaindal** [comment]
@dnsaoki2 it would be awesome if you can help, do you want me to assign you to this issue?

**Comment by dnsaoki2** [comment]
@emmaindal yes, please :smile:

**Comment by emmaindal** [comment]
closed by #8237
